src/BaGet.Aws/S3StorageOptions.cs

```diff
@@ -22,5 +22,11 @@ public class S3StorageOptions
         public bool UseInstanceProfile { get; set; }
 
         public string AssumeRoleArn { get; set; }
+
+        public bool ForcePathStyle { get; set; } = false;
+
+        public string ServiceUrl { get; set; }
```

Comment:
Could we add the data annotation attributes that @DTeuchert had used in https://github.com/loic-sharma/BaGet/pull/272?

```diff
-       [Required]
+       [RequiredIf(nameof(ServiceUrl), null)]
        public string Region { get; set; }

+       [RequiredIf(nameof(Region), null)]
        public string ServiceUrl { get; set; }
```

Reply:
Yes in my understanding we need a service url for a private / on premise solutions  or a region.